OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs

```diff
@@ -24,36 +25,42 @@ public class CreateMPPlayersInfo : TraitInfo<CreateMPPlayers>, ICreatePlayersInf
 		/// Returns a list of GameInformation.Players that matches the indexing of ICreatePlayers.CreatePlayers.
 		/// Non-playable players appear as null in the list.
 		/// </summary>
-		void ICreatePlayersInfo.CreateServerPlayers(MapPreview map, Session lobbyInfo, List<GameInformation.Player> players)
+		void ICreatePlayersInfo.CreateServerPlayers(MapPreview map, Session lobbyInfo, List<GameInformation.Player> players, MersenneTwister playerRandom)
 		{
+			var factions = map.Rules.Actors["world"].TraitInfos<FactionInfo>().ToArray();
+			var bots = map.Rules.Actors["player"].TraitInfos<IBotInfo>().ToArray();
+			var assignSpawnLocations = map.Rules.Actors["world"].TraitInfoOrDefault<IAssignSpawnPointsInfo>();
```

Comment:
Can we pull `map.Rules.Actors["world"]` into a local variable?

Reply:
Note that this is two "world"s and a "player". Do you want a `worldInfo` that we use twice and then keep `map.Rules.Actors["player"]`, or should I pull out both even though `playerInfo` would only be used once?

IMO both these options are worse than just using `map.Rules.Actors["world"]` twice.

Reply:
I wouldn't do it for player, but for world it seems like saving a lookup, so I'd go for it. If you're worried about how things look, `bots` is only used further down and can be defined there. (Actually, why was this moved up in the first place?)

Reply:
Ok, done.